buildings/gui/alter_building_relationships.py

```diff
@@ -1408,3 +1459,120 @@ def select_row_in_tbl_added(self, id_bulk):
+
+        self.lst_existing.clear()
+        self.lst_bulk.clear()
+
+        self.edit_dialog.edit_geometry()
+        self.edit_dialog.show()
+        self.change_instance = self.edit_dialog.get_change_instance()
+
+    def canvas_edit_attribute(self):
+        """
+            When edit outline radio button toggled
+        """
+        for val in [str(layer.id()) for layer in iface.legendInterface().layers()]:
+            if 'existing_subset_extracts' in val:
+                self.lyr_existing.removeSelection()
+
+        self.lst_existing.clear()
+        self.lst_bulk.clear()
+
+        self.edit_dialog.show()
+        self.edit_dialog.edit_attribute()
+        self.change_instance = self.edit_dialog.get_change_instance()
+
+    def circle_tool_clicked(self):
+        circle_tool.setup_circle(self)
+
+    def edit_cancel_clicked(self):
+        if len(QgsMapLayerRegistry.instance().mapLayersByName('bulk_load_outlines')) > 0:
+            if isinstance(self.change_instance, bulk_load_changes.EditAttribute):
+                try:
+                    self.lyr_bulk_load.selectionChanged.disconnect(self.change_instance.selection_changed)
+                except TypeError:
+                    pass
+            elif isinstance(self.change_instance, bulk_load_changes.EditGeometry):
+                try:
+                    self.lyr_bulk_load.geometryChanged.disconnect()
+                except TypeError:
+                    pass
+            elif isinstance(self.change_instance, bulk_load_changes.AddBulkLoad):
+                try:
+                    self.lyr_bulk_load.featureAdded.disconnect()
+                except TypeError:
+                    pass
+                try:
+                    self.lyr_bulk_load.featureDeleted.disconnect()
+                except TypeError:
+                    pass
+                try:
+                    self.lyr_bulk_load.geometryChanged.disconnect()
+                except TypeError:
+                    pass
+                if self.polyline:
+                    self.polyline.reset()
+                if isinstance(self.circle_tool, PointTool):
+                    self.circle_tool.canvas_clicked.disconnect()
+                    self.circle_tool.mouse_moved.disconnect()
+                    self.circle_tool.deactivate()
+                iface.actionPan().trigger()
+
+        iface.actionCancelEdits().trigger()
+
+        QgsMapLayerRegistry.instance().layerWillBeRemoved.disconnect(self.layers_removed)
+        self.edit_dialog.remove_territorial_auth()
+        QgsMapLayerRegistry.instance().layerWillBeRemoved.connect(self.layers_removed)
+
+        self.toolbar_setup()
+
+        for val in [str(layer.id()) for layer in iface.legendInterface().layers()]:
+            if 'existing_subset_extracts' in val:
+                self.lyr_existing.removeSelection()
+            if 'bulk_load_outlines' in val:
+                self.lyr_bulk_load.removeSelection()
+
+        self.lst_existing.clear()
```

Comment:
:smile: I think the lists and table should've been quite clear when this function runs. So I would remove these three lines. But there's nothing wrong to leave them!!